kura/org.eclipse.kura.protocol.can.test/src/main/java/org/eclipse/kura/protocol/can/test/CanSocketTest.java

```diff
@@ -152,13 +136,11 @@ public boolean doCanTest() {
             } catch (InterruptedException e) {
             }
         } else {
-            s_logger.info("Wait for a request");
+            logger.info("Wait for a request");
             try {
-
-                cm = this.m_canConnection.receiveCanMessage(-1, 0x7FF);
-
+                cm = this.canConnection.receiveCanMessage(-1, 0x7FF);
```

Comment:
@amitjoy I believe you are right on requiring volatile on all the fields that are modified in the ``updated`` method the way it's written now. However this ``updated`` method does not follow our tipical pattern of:

1. Stop all the threads
2. Modify internal state
3. Restart the threads (the run method will see the last modified stated)

But there is a more fundamental flaw. This OSGi (integration test) will not even run from Maven or Eclipse because it does not use the approach described in [1].

In my opinion the test will be ignored on the Mac (because of the assumption in the ``setup`` method) and will always succeed without testing anything because this.canConnection is always null in startCanTestThread.

@markoer what do you think?

[1] https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.core.test/src/main/java/org/eclipse/kura/core/test/ExampleTest.java#L41